File: Controllers/EventProfController.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;
using Microsoft.AspNetCore.Http;
using Microsoft.AspNetCore.Mvc;
using Microsoft.EntityFrameworkCore;
using Models;
using Api.Providers;
using Microsoft.AspNetCore.WebUtilities;
using System.Text;

namespace Controllers
{
    [Route("api/[controller]/[action]")]
    [ApiController]
    public class EventProfsController : SuperController<EventProf>
    {
        public EventProfsController(MyContext context ) : base(context)
        { }

        [HttpGet("{startIndex}/{pageSize}/{sortBy}/{sortDir}/{title}/{color}/{resizable}/{month}/{year}/{idUser}")]
        public async Task<IActionResult> GetAll(int startIndex, int pageSize, string sortBy, string sortDir, string title, string color, string resizable, int month, int year, int idUser)
        {
            var q = _context.EventProfs
                .Where(e => title == "*" ? true : e.Title.ToLower().Contains(title.ToLower()))
.Where(e => color == "*" ? true : e.Color.ToLower().Contains(color.ToLower()))
.Where(e => resizable == "*" ? true : e.Resizable.ToLower().Contains(resizable.ToLower()))
.Where(e => month == 0 ? true : e.Month == month)
.Where(e => year == 0 ? true : e.Year == year)
.Where(e => idUser == 0 ? true : e.IdUser == idUser)

                ;

            int count = await q.CountAsync();

            var list = await q.OrderByName<EventProf>(sortBy, sortDir == "desc")
                .Skip(startIndex)
                .Take(pageSize)
                
                .Select(e => new 
{
id = e.Id,
title = e.Title,
start = e.Start,
end = e.End,
color = e.Color,
draggable = e.Draggable,
resizable = e.Resizable,
month = e.Month,
year = e.Year,
user = e.User.Nom,
idUser = e.IdUser,

})
                .ToListAsync()
                ;

            return Ok(new { list = list, count = count });
        }
    }
}

File: laravel/app/Http/Controllers/EventProfController.php
<?php

namespace App\Http\Controllers;

use App\EventProf;
use Illuminate\Http\Request;

class EventProfController extends SuperController
{
    public function __construct(EventProf $model)
    {
        parent::__construct($model);
    }

    public function getAll(int $startIndex, int $pageSize, string $sortBy, string $sortDir
        , int $year, int $month, int $idUser, string $title) // : Collection
    {
        $matchThese = [ ];

        $year == 0 ?: array_push($matchThese, ['year', $year]);
        $month == 0 ?: array_push($matchThese, ['month', $month]);
        $idUser == 0 ?: array_push($matchThese, ['idUser', $idUser]);
        $title == '*' ?: array_push($matchThese, ['title', 'LIKE', "%{$title}%"]);


        $q = $this->_context
            ->where($matchThese)
            ->orderBy($sortBy, $sortDir);

        $count = $q->count();

        $list = $q->skip($startIndex)
            ->take($pageSize)
            ->with(['user:id,nom,prenom'])
            ->get()
            ;

        return ['list' => $list, 'count' => $count];
    }


    public function getByYearMonth(int $year, int $month, int $idUser) // : Collection
    {
        $matchThese = [ ];

        $year == 0 ?: array_push($matchThese, ['year', $year]);
        $month == 0 ?: array_push($matchThese, ['month', $month]);
        $idUser == 0 ?: array_push($matchThese, ['idUser', $idUser]);


        $list = $this->_context
            ->where($matchThese)
            ->with(['user:id,nom,prenom'])
            ->get()
            ;

        return $list;
    }
}
